core/src/block_creation_loop.rs

```diff
@@ -310,15 +483,20 @@ fn start_leader_retry_replay(
     parent_slot: Slot,
     skip_timer: Instant,
     ctx: &LeaderContext,
+    metrics: &mut BlockCreationLoopMetrics,
 ) -> Result<(), StartLeaderError> {
     let my_pubkey = ctx.my_pubkey;
     let timeout = block_timeout(leader_slot_index(slot));
     while !timeout.saturating_sub(skip_timer.elapsed()).is_zero() {
-        match maybe_start_leader(slot, parent_slot, ctx) {
+        match maybe_start_leader(slot, parent_slot, ctx, metrics) {
             Ok(()) => {
                 return Ok(());
             }
             Err(StartLeaderError::ReplayIsBehind(_)) => {
+                metrics
+                    .replay_is_behind_count
+                    .fetch_add(1, Ordering::Relaxed);
+
```

Comment:
@AshwinSekar could this potentially be triggered a lot for the same slot if you have a situation where:

1. The parent slot is X, bank is not frozen
2. There is highest_frozen_slot > X so we don't block on the wait condition here: https://github.com/anza-xyz/alpenglow/blob/ccf1550c5ce9fe399288610f5cdea33279ad4c05/core/src/block_creation_loop.rs#L519